### PR TITLE
Solve #30 - Redundant hint messages

### PR DESCRIPTION
**Pull request by k0pernicus**
Add a simple change to avoid redundant hint message, installing Nim for the first time.
To do that, I added a new switch called 'firstInstall', that will enable a boolean value in `CliParams` (cliparams.nim).
This boolean value will be check in `switcher.nim`. If the condition is verified, the program will not print out hints about non-existing binaries in local path.

Simple resolution for #30

### CODE REVIEW

File: src/choosenim/cliparams.nim [hunk 1]
```diff
@@ -56,6 +57,10 @@ Options:
                         installed. Default: ~/.choosenim.
   --nimbleDir:<dir>     Specify the Nimble directory where binaries will be
                         placed. Default: ~/.nimble.
+  --firstInstall       Specify you are trying to install Nim for the first
```
Comment (dom96) [inline src/choosenim/cliparams.nim]: Please shorten this and make sure it's aligned with the rest of the help (you're missing a single space before ``Specify`` :))

This shouldn't be used by users, so just write "Used by install script."

File: src/choosenim/switcher.nim [hunk 2]
```diff
@@ -111,8 +111,11 @@ proc writeProxy(bin: string, params: CliParams) =
   # Check whether this is in the user's PATH.
   let fromPATH = findExe(bin)
   if fromPATH == "":
-    display("Hint:", "Binary '$1' isn't in your PATH. Add '$2' to your PATH." %
-            [bin, params.getBinDir()], Warning, HighPriority)
+    # If the binary does not exists in the bin directory, and the option
+    # firstInstall is not set, display an hint to indicate the solution.
+    if not params.firstInstall:
```
Comment (dom96) [inline src/choosenim/switcher.nim]: Reduce this nesting and remove the comment (the code is self-explanatory anyway).

File: src/choosenim/cliparams.nim (hunk 1 above)
Comment (k0pernicus) [reply src/choosenim/cliparams.nim]: Done!

File: src/choosenim/switcher.nim (hunk 2 above)
Comment (k0pernicus) [reply src/choosenim/switcher.nim]: Done!

**Comment by k0pernicus** [comment]
@dom96 I updated the PR ;-)
Actually, I am not convinced about the nesting reduction but...

File: src/choosenim/cliparams.nim [hunk 3]
```diff
@@ -56,6 +57,8 @@ Options:
                         installed. Default: ~/.choosenim.
   --nimbleDir:<dir>     Specify the Nimble directory where binaries will be
                         placed. Default: ~/.nimble.
+  --firstInstall        Specify user friendly and less redundant
```
Comment (dom96) [inline src/choosenim/cliparams.nim]: I think it's enough to just write "Used by install script" here. No need for the user to know what this does.

**Comment by dom96** [comment]
> Actually, I am not convinced about the nesting reduction but...

It is a shame that you need to edit the ``elif`` condition, but I do still prefer it.

**Comment by k0pernicus** [comment]
No problem :-)

File: src/choosenim/cliparams.nim (hunk 3 above)
Comment (k0pernicus) [reply src/choosenim/cliparams.nim]: Done!

**Comment by dom96** [comment]
Brilliant, thank you!

**Pull request merged**